R/addLDA.R

```diff
@@ -0,0 +1,105 @@
+#' Latent Dirichlet Allocation
+#'
+#' These functions perform Latent Dirichlet Allocation on data stored in a 
+#'  \code{\link[TreeSummarizedExperiment:TreeSummarizedExperiment-class]{TreeSummarizedExperiment}}
+#'  object.
+#' 
+#' @param x a \code{\link[TreeSummarizedExperiment:TreeSummarizedExperiment-class]{TreeSummarizedExperiment}}
+#'  object.
+#'   
+#' @param k Integer, number of latent vectors/topics 
+#'   (By default: \code{2}).
+#' 
+#' @param name The name to be used to store the result in the reducedDims of the
+#'  output (By default: \code{name = "LDA"}).
+#'  
+#' @param assay.type a single \code{character} value for specifying which
+#'   assay to use for LDA ordination.
+#' 
+#' @param ... optional arguments.
+#' 
+#' @return 
+#' For \code{getLDA}, the ordination matrix with feature loadings matrix
+#'  as attribute \code{"loadings"}.
+#'  
+#' For \code{addLDA}, a \code{\link[TreeSummarizedExperiment:TreeSummarizedExperiment-class]{TreeSummarizedExperiment}}
+#'  object is returned containing the ordination matrix in reducedDims(..., name)
+#'  with feature loadings matrix as attribute \code{"loadings"}.
+#'  
+#' @name addLDA
+#' 
+#' @examples
+#' data(GlobalPatterns)
+#' 
+#' # Reduce the number of features 
+#' tse <- agglomerateByPrevalence(GlobalPatterns, rank="Phylum", prevalence=0.99, update.tree = TRUE)
+#' 
+#' # Run LDA and add the result to reducedDim(tse, "LDA")
+#' tse <- addLDA(tse)
+#' 
+#' # Extract feature loadings
+#' loadings <- attr(reducedDim(tse, "LDA"), "loadings")
+#' head(loadings)
+NULL
+
+#' @rdname addLDA
+#' @export
+setGeneric("getLDA", signature = c("x"),
+           function(x, ...)
+             standardGeneric("getLDA"))
+
+#' @rdname addLDA
+#' @export
+setGeneric("addLDA", signature = c("x"),
+           function(x, ...)
+             standardGeneric("addLDA"))
+
+#' @export
+#' @rdname addLDA
+setMethod("getLDA", "SummarizedExperiment",
+    function(x, k=2, assay.type = "counts", ...){
```

Comment:
k = 2 instead of k=2 --> use similar coding-style (you have assay.type = "counts" already)